src/main/kotlin/com/stefan_grafberger/streamdq/anomalydetection/strategies/windowfunctions/IntervalNormalAggregate.kt

```diff
@@ -0,0 +1,86 @@
+package com.stefan_grafberger.streamdq.anomalydetection.strategies.windowfunctions
+
+import com.stefan_grafberger.streamdq.anomalydetection.model.dto.NormalStrategyResultDto
+import com.stefan_grafberger.streamdq.checks.AggregateConstraintResult
+import org.apache.flink.api.common.functions.AggregateFunction
+import org.apache.flink.api.java.tuple.Tuple3
+import kotlin.math.pow
+import kotlin.math.sqrt
+
+@Deprecated("Use OnlineNormalAggregate with sliding window replace this")
+class IntervalNormalAggregate(
+        private val lowerDeviationFactor: Double? = 3.0,
+        private val upperDeviationFactor: Double? = 3.0,
+        private val includeInterval: Boolean = false,
+        private val waterMarkInterval: Pair<Long, Long>?)
+    : AggregateFunction<AggregateConstraintResult, Tuple3<Double, Double, Long>, NormalStrategyResultDto> {
+
+    private var currentValue = 0.0
+    private var currentTimeStamp = 0L
+
+    /**
+     * accumulator used to preserve current mean of aggregate values,
+     * current variance and count of the elements
+     * Tuple3
+     * (currentMean f0, currentVariance f1, count f2)
+     */
+    override fun createAccumulator(): Tuple3<Double, Double, Long> {
+        return Tuple3<Double, Double, Long>(0.0, 0.0, 0L)
+    }
+
+    override fun add(aggregateConstraintResult: AggregateConstraintResult,
+                     acc: Tuple3<Double, Double, Long>): Tuple3<Double, Double, Long> {
+        currentValue = aggregateConstraintResult.aggregate!!
+        currentTimeStamp = aggregateConstraintResult.timestamp
+
+        if (!includeInterval && waterMarkInterval != null) {
+            val (startInterval, endInterval) = waterMarkInterval
+            if (currentTimeStamp !in startInterval..endInterval) {
+                acc.f0 = if (acc.f2 == 0L) {
```

Comment:
What do you think about introducing new variables here at the beginning, like currentMean, currentVariante, and count, do make the following code more readable, and then only setting f0, f1, and f2 to the new values in the end?

Reply:
Or does Flink also allow using small data classes to basically have named tuples instead of tuples in the accumulator? Or is the usage of tuples here required/preferred for some reason like performance?

Reply:
Good idea! Like the 2nd way by using a data class, found Flink allows that since createAccumulator just returns a general type called ACC, which will make it more readable this way, but I prefer only do this for **online normal** since I marked deprecated for **interval normal** aggregate function

Reply:
Oh, I only saw now that this is deprecated. If it is already deprecated, we should remove this class before anyone can start using it.